### DISCUSSION TITLE
How to customize LaTeX quotes in headers

### DISCUSSION THREAD

**Discussion by TomBener** [jgm/pandoc#9793] [Q&A]
I'm trying to customize the LaTeX quotes in headers with a Lua filter. Here's the filter I've written:

```lua
function Header(header)
  local text = pandoc.utils.stringify(header.content)
  local pdftext = text:gsub('"([^"]+)"', '“%1”')
  local latextext = text:gsub('"([^"]+)"', '``%1\'\'')

  header.content = pandoc.RawInline('latex', latextext .. '}{' .. pdftext)
  return header
end
```

When I run this filter, the output LaTeX from Markdown text `# Quotation "marks" in the header` is as follows:

```tex
\section{\texorpdfstring{Quotation “marks” in the title}{Quotation “marks” in the title}{}}\label{quotation-marks-in-the-title}
```

However, I'm expecting the output to be:

```tex
\section{\texorpdfstring{Quotation ``marks'' in the title}{Quotation “marks” in the title}\label{quotation-marks-in-the-title}
```

I've tried many ways to get the expected output but have not been successful. Any help or guidance would be greatly appreciated.

Also, I've read the related source code in [LaTeX writer](https://github.com/jgm/pandoc/blob/74f533396a71651471de982b5a81505d7ec4e3d0/src/Text/Pandoc/Writers/LaTeX.hs#L733C1-L737C52), which adds `\texorpdfstring` when the header is not `plain`. I wonder what kind of text is `plain`, is it possible to treat headers with quotes as _not_ plain?

**Comment by jgm** [comment]
What is the problem with `\texorpdfstring{Quotation “marks” in the title}`?  Does it raise an error?

**Reply by TomBener** [reply]
There is no problem for this example as it is English. But if a document contains both English and Chinese texts, with the `ctex` package, `\texorpdfstring{Quotation “marks” in the title}` would be interpreted as the full width symbol, which is undesired. The solution for the scenario is to use `\texorpdfstring{Quotation ``marks'' in the title}{Quotation “marks” in the title}` for English, while for Chinese, it is just `“中文引号”`. This is an old [issue](https://github.com/jgm/pandoc/issues/5812) that has been discussed.

**Reply by TomBener** [reply]
Without using Lua filter, I came up with an alternative way to enable the `csquotes=true` option. By running the command:

```sh
pandoc test.md -o test.tex -M csquotes=true
```

We can get the following output:

```tex
\section{\texorpdfstring{Quotation \enquote{marks} in the
header}{Quotation ``marks'' in the header}}\label{quotation-marks-in-the-header}
```

But the second argument of `\texorpdfstring` is not expected. I wish it would be `Quotation “marks” in the header` instead of `Quotation ``marks'' in the header`. Can you help guide how to make this minor change?

**Reply by jpcirrus** [reply]
Is https://github.com/josephwright/csquotes/issues/26#issuecomment-554423422 of any assistance?

**Reply by TomBener** [reply]
Thanks for your comment, I'll give it a try.
